### DISCUSSION TITLE
How can I limit the input of my LineEdit component to only time?

### DISCUSSION THREAD

**Discussion by leessmin** [slint-ui/slint#4154] [General]
I am a web programmer. It is my first time to come into contact with this native development model. The problem I am encountering now is how to input the content in my LineEdit into the time format, such as 16:00.，
![image](https://github.com/slint-ui/slint/assets/101449039/606a3e6a-909c-4232-bc33-fb1356088609)

**Comment by Horbin-Magician** [comment]
I think you might need to customize a component instead of using the LineEdit.

**Reply by leessmin** [reply]
thank you for your help，But how do I make a component myself，Can I send you a doc，think

**Reply by Horbin-Magician** [reply]
I'm only a beginner too. You can wait for major contributors to provide more authoritative answers.

I just give you an idea of me:
1、Use a LineEdit (or others) to show the time, but configure it to reject all keyboard input.
2、Use a FocusScope handle keyboard input and update the content of the LineEdit accordingly.

You can add me on QQ: 2240045751, and communicate in Chinese

**Reply by leessmin** [reply]
好哒，谢谢兄弟

**Comment by hunger** [comment]
Thank you for your question and sorry for taking so long to get back to you!

At this time you will need to do something using the `edited()` and/or `accepted()` callbacks, probably even in the business logic. We should definitely have something simpler than that. I opened #4288 so we will not forget this.

**Reply by leessmin** [reply]
Thank you for taking time out of your busy schedule to answer my questions. Thank you for your attention. I hope slint-ui gets better and better
